### ISSUE TITLE
Dramatic XGBoost accuracy degradation on Iris dataset.

### ISSUE THREAD

**Issue by razdoburdin** [dmlc/xgboost#7605]
Hi,
I have measured the XGBoost accuracy on the Iris dataset and have found dramatic degradation while switching from branch release_1.5.1 to master.

I have built xgboost from sources according to the instructions for Linux.
You can find a minimal code for reproducing the problem below:

```
import xgboost
import sklearn.datasets
import sklearn.model_selection
import sklearn.metrics

X, y = sklearn.datasets.load_iris(return_X_y=True)
Xtrn, Xtst, ytrn, ytst = sklearn.model_selection.train_test_split(
    X, y, train_size=0.8, random_state=4)

clf = xgboost.XGBClassifier(
    max_depth=3,
    learning_rate=1,
    n_estimators=10,
    silent=True,
    objective='multi:softmax',
    seed=5)
clf.fit(Xtrn, ytrn)
ypred = clf.predict(Xtst)
acc = sklearn.metrics.accuracy_score(ytst, ypred)

print('xgboost accuracy on iris:', acc)

```

If the branch "release_1.5.1" is in use, the xgboost accuracy is >0.96, but for the master branch accuracy is 0.7

**Comment by trivialfis** [comment]
Thank you for raising the issue.  Looking into it.  Interesting that if the objective is automatically configured then it works fine.

**Comment by hcho3** [comment]
Just reproduced on my end too. The behavior of `clf.predict()` is different when `objective` is switched from `multi:softmax` to `multi:softprob`.
